packages/x-date-pickers-pro/src/internal/hooks/validation/useDateTimeRangeValidation.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  useValidation,
+  Validator,
+  DateTimeValidationError,
+  validateDateTime,
+  BaseDateValidationProps,
+  TimeValidationProps,
+  ValidationProps,
+} from '@mui/x-date-pickers/internals';
+import { isRangeValid } from '../../utils/date-utils';
+import { DayRangeValidationProps } from '../../models/dateRange';
+import { DateRange } from '../../models/range';
+
+export interface DateTimeRangeComponentValidationProps<TDate>
+  extends DayRangeValidationProps<TDate>,
+    TimeValidationProps<TDate>,
+    Required<BaseDateValidationProps<TDate>> {}
+
+export const validateDateTimeRange: Validator<
+  DateRange<any>,
+  any,
+  DateTimeRangeValidationError,
+  DateTimeRangeComponentValidationProps<any>
+> = ({ props, value, adapter }) => {
+  const [start, end] = value;
+
+  // for partial input
+  if (start === null || end === null) {
+    return [null, null];
+  }
+
+  const { shouldDisableDate, ...otherProps } = props;
+
+  const dateTimeValidations: [
+    DateTimeRangeValidationErrorValue,
+    DateTimeRangeValidationErrorValue,
+  ] = [
+    validateDateTime({
+      adapter,
+      value: start,
+      props: {
+        ...otherProps,
+        shouldDisableDate: (day) => !!shouldDisableDate?.(day, 'start'),
+      },
+    }),
+    validateDateTime({
+      adapter,
+      value: end,
+      props: {
+        ...otherProps,
+        shouldDisableDate: (day) => !!shouldDisableDate?.(day, 'end'),
+      },
+    }),
+  ];
+
+  if (dateTimeValidations[0] || dateTimeValidations[1]) {
+    return dateTimeValidations;
+  }
+
+  if (!isRangeValid(adapter.utils, value)) {
+    return ['invalidRange', 'invalidRange'];
+  }
+
+  return [null, null];
+};
+
+type DateTimeRangeValidationErrorValue = DateTimeValidationError | 'invalidRange' | null;
```

Comment:
1. Should we export these types as we with various validation error types in `x-date-pickers/src/index`? Users might need them to type methods handling `onError` or typing the variables storing error itself.
2. `| null` is redundant here as it is already included in `DateTimeValidationError` as one of the common error types.